File: Develop/src/Registro.java
import java.util.ArrayList;

public class Registro {
    ArrayList<Users> Users = new ArrayList();
    ArrayList<Books> Books = new ArrayList();
    ArrayList<Books> Booksdisponibles = new ArrayList();
    ArrayList<Books> Booksrentados = new ArrayList();
    ArrayList<Users> UsersCompraron = new ArrayList();
    ArrayList<Books> BooksCompraron = new ArrayList();

    public Registro() {
    }

    public void RegistroPersonas(Users persona) {
        this.Users.add(persona);
    }

    public ArrayList<Users> getUsers() {
        return this.Users;
    }

    public void mostrarUsers() {
        for(int i = 0; i < this.Users.size(); ++i) {
            System.out.printf("%d Nombre:%s Edad:%s\n",i, ((Users)this.Users.get(i)).getName(), ((Users)this.Users.get(i)).getEdad());
        }

    }

    public void RegistroLibros(Books libro) {
        this.Books.add(libro);
        this.Booksdisponibles.add(libro);
    }

    public ArrayList<Books> getBooks() {
        return this.Books;
    }

    public void mostrarBooks() {
        for(int i = 0; i < this.Books.size(); ++i) {
            System.out.printf("Titulo:%s Autor:%s\n", ((Books)this.Books.get(i)).getName(), ((Books)this.Books.get(i)).getAutor());
        }

    }

    public void mostrarBooksDisponible() {
        for(int i = 0; i < this.Booksdisponibles.size(); ++i) {
            System.out.printf("%d Titulo:%s Autor:%s\n", i, ((Books)this.Booksdisponibles.get(i)).getName(), ((Books)this.Booksdisponibles.get(i)).getAutor());
        }

    }

    public void mostrarBooksRentados() {
        for(int i = 0; i < this.Booksrentados.size(); ++i) {
            System.out.printf("%d Titulo:%s Autor:%s\n", i, ((Books)this.Booksrentados.get(i)).getName(), ((Books)this.Booksrentados.get(i)).getAutor());
        }

    }
    public void mostrarBooksComprados() {
        for(int i = 0; i < this.BooksCompraron.size(); ++i) {

            System.out.printf("El usuario: %s de %s años compro: %s de %s\n",UsersCompraron.get(i).getName(),UsersCompraron.get(i).getEdad(),BooksCompraron.get(i).getName(),BooksCompraron.get(i).getAutor());


        }

    }

    public void rentarLibro(int opcion) {
        Books librorentado = (Books)this.Booksdisponibles.get(opcion);
        this.Booksrentados.add(librorentado);
        this.Booksdisponibles.remove(opcion);
        System.out.println("****************** ");
    }

    public void regresarLibro(int opcion) {
        Books libroRegresado =Booksrentados.get(opcion);
        this.Booksdisponibles.add(libroRegresado);
        System.out.println("Se ha regresado Exitosamente");
    }

    public void comprarLibro(int opcion,int opcion2) {
        Books librocomprado = Booksdisponibles.get(opcion);
        Users usercomprador=Users.get(opcion2);
        UsersCompraron.add(usercomprador);
        BooksCompraron.add(librocomprado);
        for(int i = 0; i < this.Books.size(); ++i) {
            if (librocomprado == this.Books.get(i)) {
                this.Books.remove(i);
            }
        }

        this.Booksdisponibles.remove(opcion);
        System.out.println("Su compra ha sido exitosa:)");
    }
}

File: Registro.py
class Registro:
    def __init__(self):
        self.users = []
        self.books = []
        self.books_disponibles = []
        self.books_rentados = []
        self.users_compraron = []
        self.books_compraron = []

    def registro_personas(self, persona):
        self.users.append(persona)

    def mostrar_users(self):
        for i, user in enumerate(self.users):
            print(f"{i} Nombre: {user.getName()} Edad: {user.getEdad()}")

    def registro_libros(self, libro):
        self.books.append(libro)
        self.books_disponibles.append(libro)

    def mostrar_books(self):
        for book in self.books:
            print(f"Titulo: {book.getName()} Autor: {book.getAutor()}")

    def mostrar_books_disponible(self):
        for i, book in enumerate(self.books_disponibles):
            print(f"{i} Titulo: {book.getName()} Autor: {book.getAutor()}")

    def mostrar_books_rentados(self):
        for i, book in enumerate(self.books_rentados):
            print(f"{i} Titulo: {book.getName()} Autor: {book.getAutor()}")

    def mostrar_books_comprados(self):
        for i in range(len(self.books_compraron)):
            print(f"El usuario: {self.users_compraron[i].getName()} de {self.users_compraron[i].getEdad()} años compró: {self.books_compraron[i].getName()} de {self.books_compraron[i].getAutor()}")

    def rentar_libro(self, opcion):
        libro_rentado = self.books_disponibles[opcion]
        self.books_rentados.append(libro_rentado)
        del self.books_disponibles[opcion]
        print("******************")

    def regresar_libro(self, opcion):
        libro_regresado = self.books_rentados[opcion]
        self.books_disponibles.append(libro_regresado)
        print("Se ha regresado Exitosamente")

    def comprar_libro(self, opcion, opcion2):
        libro_comprado = self.books_disponibles[opcion]
        user_comprador = self.users[opcion2]
        self.users_compraron.append(user_comprador)
        self.books_compraron.append(libro_comprado)
        self.books.remove(libro_comprado)
        del self.books_disponibles[opcion]
        print("Su compra ha sido exitosa:)")
